src/utils/reducer-store.js

```diff
@@ -23,26 +23,26 @@ class ReducerStore {
      * value of `defaultReducer` is used for aggregation.
      *
      * @public
-     *
-     * @param {string} [reducer='sum'] name of the default reducer. It picks up the definition from store by doing name
-     *      lookup. If no name is found then it takes `sum` as the default reducer.
-     *
-     * @return {ReducerStore} instance of the singleton store in page.
+     * @param {string} [reducer='sum'] - The name of the default reducer. It picks up the definition from store by doing
+     * name lookup. If no name is found then it takes `sum` as the default reducer.
+     * @return {ReducerStore} Returns instance of the singleton store in page.
      */
-    defaultReducer (...params) {
-        if (params.length) {
-            let reducer = params[0];
-            if (typeof reducer === 'function') {
-                this.store.set('defReducer', reducer);
-            } else if (typeof reducer === 'string') {
-                if (Object.keys(fnList).indexOf(reducer) !== -1) {
-                    this.store.set('defReducer', fnList[reducer]);
-                }
-            }
-            return this;
+    defaultReducer (reducer) {
+        if (!reducer) {
+            return this.store.get('defReducer');
```

Comment:
What was the problem with the last implementation? The last one looks semantically and logically more correct.

Reply:
1. The defaultReducer function expects only one param and behaves as getter and setter according to the presence of argument. So, we can easily check the getter and setter behaviour by the `reducer` argument's `undefined` state instead of params length and the `reducer` variable name is more descriptive than the params length checking.
2. The old one only checks the `function` and `string` type, does nothing for other types and which makes user unaware about this, so to solve this it could throw a warning/error for the other data types or convert that to string if the reducer is not a fucntion.

Reply:
Yes. But `!reducer` returns true for any falsy value (empty string / zero)
```javascript
fn (...params) {
    if (params.length) { }
}
```
is a much more robust check.

Reply:
Okay, I am correcting it.